app/testutils.py

```diff
@@ -0,0 +1,79 @@
+"""Utilties for unit testing.
+
+Copyright 2016-2017 Association of Universities for Research in Astronomy, Inc.
+Copyright 2014 Miguel Grinberg.
+"""
+
+from base64 import b64encode
+from collections import namedtuple
+import json
+from urllib.parse import urlsplit, urlunsplit
+
+Response = namedtuple('Response', 'status headers json')
+
+
+class TestClient():
+    """TestClient wraps Flask's/Werkzeug's built-in testing client.
+
+    The `get`, `post`, `put`, `delete` methods mirror HTTP
+    commands and return a `response` `namedtuple` with fields:
+
+    - `status`: the integer HTTP response code
+    - `header`: the HTTP response headers
+    - `json`: the return data, parse as JSON into a Python `dict` object.
+    """
+    def __init__(self, app, username='', password=''):
+        self.app = app
+        self.auth = 'Basic ' + b64encode((username + ':' + password)
+                                         .encode('utf-8')).decode('utf-8')
```

Comment:
Good question. That one comes from the O'Reilly book (https://github.com/miguelgrinberg/flasky/blob/master/tests/test_api.py#L27) and wasn't described.